### PR TITLE
Document release process

### LINKED ISSUE #510: Document release process
We've had a few teething problems and I think a few implicit assumptions getting our first release out. Let's come up with a clear release process doc, nothing long, just a list of how all the pieces fit together, and the related tasks.

* How to perform a release (just tag locally and go? or pr?)
* Generate and publicise release notes
* Register the release with GitHub's releases
* List comms tasks & templates

### PR DESCRIPTION
**Pull request by japgolly**
Closes #510

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   revise,
+   and git Publish.
+
+4. Advertise the new release in the following places:
+
+    * Reddit
+      Example: https://old.reddit.com/r/scala/comments/pc9k5y/scalajsdom_120_released_this_is_the_first_release
+
+    * Scala.js Discord
+      Example: ?
```
Comment (armanbilge) [inline RELEASING.md:39]:
```suggestion
      Example: https://discord.com/channels/632150470000902164/635668814956068864/880575154516819978
```

**Review by armanbilge — APPROVED** [review]
Thanks for writing this all out! Super important for some of us 😳

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ```
+
+
+2. Wait for the release to complete.
+   https://github.com/scala-js/scala-js-dom/actions/workflows/release.yml
+
+3. Go to GitHub releases (https://github.com/scala-js/scala-js-dom/releases),
+   edit the auto-generated draft release,
+   revise,
+   and git Publish.
```
[the remark this answers is not in the document]
Comment (japgolly) [reply RELEASING.md:31]: Whoops, was originally "hit". Thanks for correcting!!

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+How to Perform a Release
+========================
+
+1. The first step is to create a tag, push it, and let GitHub Actions do the release.
+
+    ```sh
+    # For v1.x.y
+    ver=1.2.0
```
Comment (armanbilge) [inline RELEASING.md:8]:
```suggestion
    ver=1.2.0 # change this to new version
```

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    git checkout series/1.x
+    git checkout -b topic/v$ver
+    echo "ThisBuild / version := \"$ver\" // Workaround for #504" > version.sbt
+    git add version.sbt
+    git commit -m "Setting version to v$ver"
+    git tag -s v$ver
+    git push --tags
+
+    # For v2.x onwards
+    ver=2.0.0
```
Comment (armanbilge) [inline RELEASING.md:18]:
```suggestion
    ver=2.0.0 # change this to new version
```

**Review by armanbilge — APPROVED** [review]

**Pull request merged**